src/ui.rs

```diff
@@ -0,0 +1,394 @@
+                GRAY,
+            )
+        }
+    });
+
+    let text = format!("Generations: {}", generations_passed);
+    draw_text(
+        &text,
+        screen_width * GENERATIONS_POS_X,
+        screen_height * GENERATIONS_POS_Y,
+        TEXT_SIZE,
+        BLACK,
+    );
+
+    let text = format!("FPS: {}", (1. / fps).round());
+    draw_text(
+        &text,
+        screen_width * GENERATIONS_POS_X,
+        screen_height * GENERATIONS_POS_Y + 15.,
+        TEXT_SIZE,
+        BLACK,
+    );
+
+    render_mode(
+        game,
+        (offset_x, offset_y, sq_size),
+        game_paused,
+        edit_mode,
+        generations_passed,
+        fps,
+        acc,
+    );
+}
+
+/// Renders the game based on the current mode (playing, paused, edit).
+fn render_mode(
+    game: &mut Game,
+    screen_params: (f32, f32, f32),
+    game_paused: &mut bool,
+    edit_mode: &mut bool,
+    generations_passed: &mut i32,
+    fps: &f32,
+    acc: &mut f32,
+) {
+    if *game_paused {
+        if *edit_mode {
+            run_edit_mode(game, screen_params);
+        } else {
+            run_paused_mode();
+        }
+    } else {
+        run_playing_mode(game, generations_passed, fps, acc);
+    }
+}
+
+/// Handles the edit mode where users can add or remove cells.
+fn run_edit_mode(game: &mut Game, screen_params: (f32, f32, f32)) {
+    show_edit_hud();
+    let (offset_x, offset_y, sq_size) = screen_params;
+
+    if is_mouse_button_pressed(MouseButton::Left) {
+        let (mouse_x, mouse_y) = mouse_position();
+
+        let grid_x = ((mouse_x - offset_x) / sq_size).floor() as i32;
+        let grid_y = ((mouse_y - offset_y) / sq_size).floor() as i32;
+
+        if grid_x >= 0 && grid_x < SQUARES as i32 && grid_y >= 0 && grid_y < SQUARES as i32 {
+            if game.is_cell_alive(grid_x, grid_y) {
+                game.remove_cell(grid_x, grid_y);
+            } else {
+                game.add_cell(grid_x, grid_y);
+            }
+        }
+    }
+}
+
+/// Displays the HUD for edit mode, showing instructions to the user.
+fn show_edit_hud() {
+    let screen_width = screen_width();
+    let screen_height = screen_height();
+
+    draw_text(
+        "Edit Mode",
+        screen_width * TITLE_POS_X,
+        screen_height * TITLE_POS_Y,
+        TITLE_SIZE,
+        BLACK,
+    );
+    draw_text(
+        "Click on a live cell to kill it",
+        screen_width * TEXT_POS_X,
+        screen_height * TEXT_POS_Y - 15.,
+        TEXT_SIZE,
+        BLACK,
+    );
+    draw_text(
+        "Click on a dead cell to make it alive",
+        screen_width * TEXT_POS_X,
+        screen_height * TEXT_POS_Y,
+        TEXT_SIZE,
+        BLACK,
+    );
+    draw_text(
+        "Press E to exit Edit Mode",
+        screen_width * TEXT_POS_X,
+        screen_height * TEXT_POS_Y + 15.,
+        TEXT_SIZE,
+        BLACK,
+    );
+}
+
+/// Runs the game in paused mode, displaying the pause HUD.
+fn run_paused_mode() {
+    show_pause_hud();
```

Comment:
We could remove the `run_paused_mode` function and use directly `show_paused_hud`

Reply:
Done!